hawc/apps/invitro/api.py

```diff
@@ -1,6 +1,42 @@
-from ..assessment.api import AssessmentRootedTagTreeViewset, AssessmentViewset
-from ..common.api import CleanupFieldsBaseViewSet
-from . import models, serializers
+from rest_framework import decorators, viewsets
+from rest_framework.response import Response
+
+from ..assessment.api import (
+    AssessmentLevelPermissions,
+    AssessmentRootedTagTreeViewset,
+    AssessmentViewset,
+)
+from ..assessment.models import Assessment
+from ..common.api import CleanupFieldsBaseViewSet, LegacyAssessmentAdapterMixin
+from ..common.renderers import PandasRenderers
+from ..common.views import AssessmentPermissionsMixin
+from . import exports, models, serializers
+
+
+class IVAssessmentViewset(
+    AssessmentPermissionsMixin, LegacyAssessmentAdapterMixin, viewsets.GenericViewSet
+):
+    parent_model = Assessment
+    model = models.IVEndpoint
+    permission_classes = (AssessmentLevelPermissions,)
+
+    def get_queryset(self):
+        perms = self.get_obj_perms()
+        if not perms["edit"]:
+            return self.model.objects.published(self.assessment)
+        return self.model.objects.get_qs(self.assessment)
+
+    @decorators.detail_route(
+        methods=("get",), url_path="full-export", renderer_classes=PandasRenderers
+    )
+    def full_export(self, request, pk):
+        self.set_legacy_attr(pk)
+        self.object_list = self.get_queryset()
+        export_format = request.GET.get("output", "excel")
```

Comment:
i don't think we need this; our renderer handles the export_format

Reply:
Understood, I just hardwired 'excel' into the export_format parameter.